react/week2/src/App.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect } from "react";
+import Modal from './components/Modal'
+import Header from './components/Header';
+import AddRandom from './components/AddRandom';
+import "./App.css";
+
+function Todo({ todo, index, completeTodo, removeTodo }) {
+  return (
+    <div
+      className="todo"
+      style={{ textDecoration: todo.isCompleted ? "line-through" : "" }}
+    >
+      {todo.text}
+
+      <div>
+        <button type="checkBox" onClick={() => completeTodo(index)}>Complete</button>
+        <button onClick={() => removeTodo(index)}>Delete</button>
+       </div>
+     
+    </div>
+    
+  );
+}
+
+function TodoForm({ addTodo }) {
+  const [value, setValue] = useState("");
+
+  const handleSubmit = e => {
+    e.preventDefault();
+   
+    if (!value) return;
+    addTodo(value);
+    setValue("");
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <input
+        type="text"
+        className="input"
+        placeholder="Enter activity here.."
+        value={value}
+        onChange={e => setValue(e.target.value)}
+      />
+    </form>
+  );
+}
+
+function App() {
+  const [todos, setTodos] = useState([
+    {
+      text: "Get out of bed",
+      isCompleted: false
+    },
+    {
+      text: "Brush teeth",
+      isCompleted: false
+    },
+    {
+      text: "Eat breakfast",
+      isCompleted: true
+    }
+  ]);
+
+  const addTodo = text => {
+    const newTodos = [...todos, { text }];
+    setTodos(newTodos);
+  };
+
+  const completeTodo = index => {
+    const newTodos = [...todos];
+    newTodos[index].isCompleted = true;
```

Comment:
Even though it might seem, as you are modifying the item inside of a new (copied) array, newTodos[index] still references the object from the original todos array. You can see it by logging out the item before and after the line 72:

console.log(todos[index])
newTodos[index].isCompleted = true;
console.log(todos[index])

Hence you are changing the object from the state directly, which is not recommended. Please check this comment I left on a similar issue to get an insight how to fix it:

https://github.com/cecastosic/hyf-homework/pull/12#discussion_r376589618